### DISCUSSION TITLE
How to compile Java source files directly to a .NET Framework binary?

### DISCUSSION THREAD

**Discussion by trungnt2910** [ikvmnet/ikvm#422] [Q&A] [answered]
I have downloaded some old code from the internet containing a few `.java` source files. There is no runnable `.jar` file.

Is there a way I can directly compile these files into a .NET Framework console application, or do I have to make them into `.class` or bundle them into a `.jar` first?

**Comment by wasabii** [comment]
Internally you have to compile them with a Java compiler first. But, IKVM.Net.Sdk exists to make that easier.

**Reply by trungnt2910** [reply]
Can you point to an example using that? I could only find guide for the `IkvmReference`, which requires precompiled `.class` or `.jar`.

**Reply by trungnt2910** [reply, marked as answer]
OK thanks, I found this in the tree, should have been a bit more visible though: https://github.com/ikvmnet/ikvm/blob/main/src/IKVM.NET.Sdk/README.md
